web/src/auth/localAuth.ts

```diff
@@ -0,0 +1,118 @@
+import { SubmitHandler } from '@redwoodjs/forms'
+
+// If you're integrating with an auth service provider you should delete this interface.
+// Instead you should import the type from their auth client sdk.
+export interface AuthClient {
+  login: () => User
+  logout: () => void
+  signup: () => User
+  getToken: () => string
+  getUserMetadata: () => User | null
+}
+
+export interface LoginEventInterface {
+  email: string
+}
+
+export interface LocalAuthClient {
+  login: (event: SubmitHandler<LoginEventInterface>) => void
+  logout: () => Promise<void>
+  signup: () => null
+  getToken: () => Promise<string>
+  getUserMetadata: () => Promise<User | null>
+}
+
+interface User {
+  id: string
+  name: string
+  email: string
+  role: string
+  agencyId: number
+  organizationId: number
+  createdAt: string
+  updatedAt: string
+}
+
+export const localAuthClient = {
+  logout: async () => {
+    localStorage.removeItem('local_auth_token')
+    window.location.href = '/login'
+  },
+  getToken: async () => {
+    return localStorage.getItem('local_auth_token')
+  },
+  getUserMetadata: async () => {
+    if (!localStorage.getItem('local_auth_token')) {
+      return null
+    }
+
+    console.log('local: getUserMetadata')
+    let user: User | null = null
+    // Explicitly setting the url to ensure these requests are not made outside of the local environment
+    await fetch(`http://localhost:8911/localAuth`, {
```

Comment:
See the related comment in the API-side `auth.ts` – I think the env var should be good enough to dictate whether local auth is enabled; there are plenty of other things that could go awry if we aren't careful about env vars in hosted environments.